src/Microsoft.Extensions.Http/DefaultHttpMessageHandlerBuilder.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+
+namespace Microsoft.Extensions.Http
+{
+    internal class DefaultHttpMessageHandlerBuilder : HttpMessageHandlerBuilder
```

Comment:
This needs some tests, waiting for feedback on design first

Reply:
I'm not sure about a few things:
- Abstract base class (come at me bro)
- The having the primary handler be settable? Why isn't it a ctor parameter? Is that for testing?

Reply:
> Abstract base class (come at me bro)

Per discussion this will be an interface - we think the API we need is simple and will not grow over time

> The having the primary handler be settable? 

Yes. Some of the handlers have configuration on them as well. If you care about setting those deets you need to be able to replace the handler. I think making it not settable raises more questions then answers

>Why isn't it a ctor parameter? 

Constructors don't compose. This class comes from DI anyway, that would be really complicated and not add anything.

> Is that for testing?

We'll probably use it for testing, but it's not **only** for testing